kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -85,10 +102,62 @@ public void forwardRequest(ApiMessage message) {
             throw new AssertionError("Attempt to use forwardRequest with a non-request: " + name);
         }
 
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("{}: Forwarding request: {}", channelDescriptor(), decodedFrame);
+        }
         // TODO check we've not forwarded it already
         channelContext.write(decodedFrame, promise);
     }
 
+    @Override
+    public <T extends ApiMessage> ProxyFuture<T> sendRequest(short apiVersion, ApiMessage message) {
+        checkNotClosed();
+        short key = message.apiKey();
+        var apiKey = ApiKeys.forId(key);
+        short headerVersion = apiKey.requestHeaderVersion(apiVersion);
+        var header = new RequestHeaderData()
+                .setCorrelationId(0)
```

Comment:
Maybe -1 would make debugging easier. It's not a value that the client itself can never use, but 0 is something the client always uses at least once.